controllers/component_build_controller_test.go

```diff
@@ -2016,6 +2029,88 @@ var _ = Describe("Component initial build controller", func() {
 			ensureNoPipelineRunsCreated(resourceKey)
 		})
 
+		It("should fail when no pipeline is selected based on predefined build pipeline selector", func() {
+			deleteBuildPipelineRunSelector(defaultSelectorKey)
+			selector := &buildappstudiov1alpha1.BuildPipelineSelector{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      defaultSelectorKey.Name,
+					Namespace: defaultSelectorKey.Namespace,
+				},
+				Spec: buildappstudiov1alpha1.BuildPipelineSelectorSpec{
+					Selectors: []buildappstudiov1alpha1.PipelineSelector{
+						{
+							Name: "java",
+							PipelineRef: tektonapi.PipelineRef{
+								Name:   "java-builder",
+								Bundle: defaultPipelineBundle,
+							},
+							PipelineParams: []buildappstudiov1alpha1.PipelineParam{
+								{
+									Name:  "additional-param",
+									Value: "additional-param-value-global",
+								},
+							},
+							WhenConditions: buildappstudiov1alpha1.WhenCondition{
+								Language: "java",
+							},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, selector)).To(Succeed())
+
+			deleteComponent(resourceKey)
+			component := getSampleComponentData(resourceKey)
+			component.Annotations[BuildRequestAnnotationName] = BuildRequestTriggerSimpleBuildAnnotationValue
+			Expect(k8sClient.Create(ctx, component)).Should(Succeed())
+
+			// devfile is about nodejs rather than Java, which causes failure of selecting a pipeline.
+			devfile := `
+                        schemaVersion: 2.2.0
+                        metadata:
+                            name: devfile-nodejs
+                            language: nodejs
+                    `
+			setComponentDevfile(resourceKey, devfile)
+
+			ensureNoPipelineRunsCreated(resourceKey)
+
+			component = getComponent(resourceKey)
+			Expect(component.Annotations[InitialBuildAnnotationName]).To(Equal("processed"))
+
+			// Check expected errors
+			statusAnnotation := component.Annotations[BuildStatusAnnotationName]
+			Expect(strings.Contains(statusAnnotation, "No pipeline is selected")).To(BeTrue())
+		})
+
+		// Always keep this test as the last one in current context
```

Comment:
Why do we need this condition? Could we just delete all selectors at the beginning of the test, so there would be no binding any more?